src/spiffs_nucleus.c

```diff
@@ -939,12 +942,13 @@ s32_t spiffs_object_create(
   fs->stats_p_allocated++;
 
   // write empty object index page
+  size_t const len = MIN( strlen((const char *)name), sizeof(oix_hdr.name));
   oix_hdr.p_hdr.obj_id = obj_id;
   oix_hdr.p_hdr.span_ix = 0;
   oix_hdr.p_hdr.flags = 0xff & ~(SPIFFS_PH_FLAG_FINAL | SPIFFS_PH_FLAG_INDEX | SPIFFS_PH_FLAG_USED);
   oix_hdr.type = type;
   oix_hdr.size = SPIFFS_UNDEFINED_LEN; // keep ones so we can update later without wasting this page
-  strncpy((char*)oix_hdr.name, (const char*)name, SPIFFS_OBJ_NAME_LEN);
+  strncpy((char*)oix_hdr.name, (const char*)name, len);
```

Comment:
This no longer copies the null at the end (len is strlen of name). ALso, if you know the length, safer to use memcpy....

Reply:
this is exactly what i wanted.
overwrites in oix_hdr should be prevented.
the null termination is done by:
    spiffs_page_object_ix_header oix_hdr = {.p_hdr = {0}};

Reply:
Hi, please use one of SPIFFS_DBG/SPIFFS_GC_DBG/SPIFFS_CACHE_DBG/SPIFFS_CHECK_DBG macro for debug printing instead of direct call to printf

Reply:
Please use memcpy then to indicate that you don't need the (rather obscure) behavior of strncpy.

Reply:
'c' is pretty obscure :-)
i personaly like the behaviour of strncpy because it copies only bytes up to the length given.

Reply:
Ah -- but that isn't what strncpy does that is magic!! 

memcpy copies the exact number of bytes. 
strlcpy copies the string but ensures that it doesn't overflow the destination and ensures that the destination is null terminated (by truncating the string if needed). 
strncpy copies the string (up to the buffer size) and then null fills the rest of the destination buffer.